src/main/java/gift/model/Option.java

```diff
@@ -51,4 +63,11 @@ public Product getProduct() {
     public Long getQuantity() {
         return quantity;
     }
+
+    public void setProduct(Product product) {
+        this.product = product;
+        if (product != null && !product.getOptions().contains(this)) {
```

Comment:
```suggestion
        if (product != null && !product.contains(this)) {
```
점(.) 을 하나만 찍을 수 있게 위와 같이 변경해보면 어떨까요?
그러면 좀 더 객체지향적인 코드가 될 것 같아요~